# Future Sales Prediction

### import relvant libraries and the dataset

In [1]:
import pandas as pd
import numpy as np

In [3]:
raw_data = pd.read_csv('sales.csv')
raw_data.head(10)

TV  Radio  Newspaper  Sales
0  230.1   37.8       69.2   22.1
1   44.5   39.3       45.1   10.4
2   17.2   45.9       69.3   12.0
3  151.5   41.3       58.5   16.5
4  180.8   10.8       58.4   17.9
5    8.7   48.9       75.0    7.2
6   57.5   32.8       23.5   11.8
7  120.2   19.6       11.6   13.2
8    8.6    2.1        1.0    4.8
9  199.8    2.6       21.2   15.6

### Preprocessing

In [4]:
raw_data.isnull().sum()

TV           0
Radio        0
Newspaper    0
Sales        0
dtype: int64

In [5]:
raw_data.duplicated().sum()

0

In [17]:
import plotly.express as px
import plotly.graph_objects as go

figure = px.scatter(data_frame=raw_data, x='Sales', y='TV', size='TV', trendline='ols')
figure.show()

In [16]:
figure = px.scatter(data_frame=raw_data, x='Sales', y='Radio', size='Radio', trendline='ols')
figure.show()

In [18]:
figure = px.scatter(data_frame=raw_data, x='Sales', y='Newspaper', size='Newspaper', trendline='ols')
figure.show()

In [19]:
inputs = raw_data.iloc[:,:-1].values
targets = raw_data.iloc[:,-1].values

In [20]:
print(inputs.shape)
print(targets.shape)

(200, 3)
(200,)


In [21]:
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(inputs, targets, test_size=0.1, random_state=0)

### Linear Regression Model

In [22]:
from sklearn.linear_model import LinearRegression
model_1 = LinearRegression()
model_1.fit(x_train, y_train)

LinearRegression()

In [23]:
y_pred = model_1.predict(x_test)

In [24]:
from sklearn.metrics import r2_score
accuracy = r2_score(y_test, y_pred)
print('the accuracy of linear regression model is {:.2f}%'.format(accuracy*100))

the accuracy of linear regression model is 81.40%


### SVR Model

In [26]:
from sklearn.svm import SVR
model_2 = SVR(kernel='rbf')
model_2.fit(x_train, y_train)

SVR()

In [28]:
y_pred = model_2.predict(x_test)

In [29]:
accuracy = r2_score(y_test, y_pred)
print('the accuracy of SVR model is {:.2f}%'.format(accuracy*100))

the accuracy of SVR model is 83.60%


### Decision tree regression model

In [33]:
from sklearn.tree import DecisionTreeRegressor
model_3 = DecisionTreeRegressor(random_state=0)
model_3.fit(x_train,y_train)

DecisionTreeRegressor(random_state=0)

In [34]:
y_pred = model_3.predict(x_test)

In [35]:
accuracy = r2_score(y_test, y_pred)
print('the accuracy of Decision tree regression model is {:.2f}%'.format(accuracy*100))

the accuracy of Decision tree regression model is 83.40%


### Random forest regression model

In [44]:
from sklearn.ensemble import RandomForestRegressor
model_4 = RandomForestRegressor(n_estimators=10,random_state=0)
model_4.fit(x_train,y_train)

RandomForestRegressor(n_estimators=10, random_state=0)

In [45]:
y_pred = model_4.predict(x_test)

In [46]:
accuracy = r2_score(y_test, y_pred)
print('the accuracy of Random forest regression model is {:.2f}%'.format(accuracy*100))

the accuracy of Random forest regression model is 93.97%


#### from training the data by more than regression models  we observe that the random forest regression is the best model with accuracy 93.97%
